compiler/rustc_typeck/src/collect/type_of.rs

```diff
@@ -29,6 +29,48 @@ pub(super) fn opt_const_param_of(tcx: TyCtxt<'_>, def_id: LocalDefId) -> Option<
         let parent_node = tcx.hir().get(parent_node_id);
 
         match parent_node {
+            Node::Ty(hir_ty @ Ty { kind: TyKind::Path(QPath::TypeRelative(_, segment)), .. }) => {
```

Comment:
This is pre-existing to some extent, but I think this code could use a few more comments. It'd be nice to give an example of the Rust code you are matching against and explain what each variable represents in terms of that code.

I'm a bit confused, for example, what the `def_id` here refers to -- is it part of the impl? the trait? 

What is the `parent_node` -- the `type Foo<const C> = ...` in the impl? I... guess so? Can it also be in a trait (e.g., with a default value?)

Reply:
I added some comments that will hopefully remove some of the confusion, I'm not super confident in them because I'm still pretty new to rustc but I think they're correct haha

Reply:
Thanks!